app/components/agent/Index/row.js

```diff
@@ -22,30 +28,69 @@ class AgentRow extends React.Component {
   };
 
   render() {
-    const iconClassName = classNames('circle', {
-      'bg-lime': this.props.agent.connectionState === 'connected',
-      'bg-gray': this.props.agent.connectionState === 'disconnected' || this.props.agent.connectionState === 'never_connected',
-      'bg-orange': this.props.agent.connectionState !== 'connected' && this.props.agent.connectionState !== 'disconnected' && this.props.agent.connectionState !== 'never_connected'
-    });
+    const { agent } = this.props;
+
+    const iconClassName = classNames(
+      'circle',
+      getColourForConnectionState(agent.connectionState, 'bg-')
+    );
 
     let metaDataContent = 'No metadata';
-    if (this.props.agent.metaData.length > 0) {
-      metaDataContent = this.props.agent.metaData.sort().join(' ');
+    if (agent.metaData.length > 0) {
+      metaDataContent = agent.metaData.sort().join(' ');
     }
 
     return (
       <Panel.Row>
         <div className="flex">
-          <div className="pr3 pt1">
-            <div className={iconClassName} style={{ width: 12, height: 12 }} />
+          <div
+            className="pr3 pt1"
+            title={getLabelForConnectionState(agent.connectionState)}
+          >
+            {
+              agent.job
```

Comment:
This bit of JS is breaking my brain a little :P

Perhaps it's better suited to another function in this component?